### ISSUE TITLE
Handle Overflow Errors

### ISSUE THREAD

**Issue by holygits** [RustPython/RustPython#276]
In many places there is a cast from a python int type to some rust primitive int type e.g. `usize` with `to_usize()` or similar.
This is frequently handled in place with .`unwrap()`.

Example of the current behaviour
```
>>>>> [1,2,3][10000000000000000000000000000000000]
thread 'main' panicked at 'called `Option::unwrap()` on a `None` value', src/libcore/option.rs:345:21
note: Run with `RUST_BACKTRACE=1` environment variable to display a backtrace.
This will panic when there are overflows or invalid values.
```
One solution maybe to wrap the conversion functions with the appropriate handling to avoid duplicating the logic everywhere.

**Comment by windelbouwman** [comment]
I checked what cpython does. It raises this `IndexError`:

`IndexError: cannot fit 'int' into an index-sized integer`

We could do the same!

**Comment by vokar97** [comment]
The issue can be closed as, I believe, #323 solves it.
